core/src/tx/signer.rs

```diff
@@ -1,4 +1,4 @@
-// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
```

Comment:
nit: Should we rename `Singer` -> `SignerT` for consistency with `AddrT` / `PayloadT`?
I've also seen it imported as `SignerT` for testing

Reply:
Good question! There are probably a bunch of other traits here that we could, if we wanted, add `T` on the end of! In this PR I just standardised the "address/payload" traits to have a `T` but maybe we should go further! What do you reckon? (miight be better to do a full standarisation as a separate PR but I'd have to check hoiw many traits we have left to add the T too :D)

Reply:
I had a look at the other traits we have and now I'm not sure which way is best! I liked the standardising here but also think that adding `T` onto everything feels a bit naff (and it's not what Rust does on any of the basic traits).

Lemme have a play; maybe I should undo the `T` thing and do something else..

Reply:
(I'll sort this out in a small separate PR anyways :))